bin/modules/calculate_properties.py

```diff
@@ -345,3 +349,590 @@ def getStructuresFromOptimization(outputfolder, FrameworkName) -> list:
+
+    with open(os.path.join(output_folder, FrameworkName), "w") as fhandle:
+        for line in generator.line_iter(input_dict):
+            fhandle.write(f"{line}\n")
+
+
+def get_forces(FrameworkName, output_folder):
+    """ Get the CP2K forces from the output file in atomic units [Hartree/a.u.]
+
+    Parameters
+    ----------
+    FrameworkName : str
+        Name of the framework
+    output_folder : str
+        Path to the output folder
+
+    Returns
+    -------
+    forces : np.ndarray
+        Nx3 Array of the forces in atomic units [a.u.]
+    """
+
+    with open(os.path.join(output_folder, f"{FrameworkName}-forces-1_0.xyz"), "r") as f:
+        lines = f.read().splitlines()
+
+    forces = []
+
+    for line in lines[4:-1]:
+        forces.append([float(i) for i in line.split()[3:]])
+
+    return np.array(forces)
+
+
+def get_pol_tensor(file_name, output_folder, symmetrize=False):
+    with open(os.path.join(output_folder, file_name), "r") as f:
+        lines = f.read().splitlines()
+
+    pol_au = np.zeros((3, 3))
+    pol_angs = np.zeros((3, 3))
+
+    for i, line in enumerate(lines):
+        if "POLARIZABILITY TENSOR (atomic units):" in line:
+            t1 = lines[i+1].split()[1:]
+            t2 = lines[i+2].split()[1:]
+            t3 = lines[i+3].split()[1:]
+            pol_au = np.array([t1, t2, t3], dtype=float)
+
+        if "POLARIZABILITY TENSOR (Angstrom^3):" in line:
+
+            t1 = lines[i+1].split()[1:]
+            t2 = lines[i+2].split()[1:]
+            t3 = lines[i+3].split()[1:]
+            pol_angs = np.array([t1, t2, t3], dtype=float)
+
+    # The output of CP2K is out of order. We need to reorder it
+    pol_au_order = np.zeros((3, 3))
+    pol_au_order[0, 0] = pol_au[0, 0]
+    pol_au_order[1, 1] = pol_au[0, 1]
+    pol_au_order[2, 2] = pol_au[0, 2]
+    pol_au_order[0, 1] = pol_au[1, 0]
+    pol_au_order[0, 2] = pol_au[1, 1]
+    pol_au_order[1, 2] = pol_au[1, 2]
+    pol_au_order[1, 0] = pol_au[2, 0]
+    pol_au_order[2, 0] = pol_au[2, 1]
+    pol_au_order[2, 1] = pol_au[2, 2]
+
+    po_angs_order = np.zeros((3, 3))
+
+    po_angs_order[0, 0] = pol_angs[0, 0]
+    po_angs_order[1, 1] = pol_angs[0, 1]
+    po_angs_order[2, 2] = pol_angs[0, 2]
+    po_angs_order[0, 1] = pol_angs[1, 0]
+    po_angs_order[0, 2] = pol_angs[1, 1]
+    po_angs_order[1, 2] = pol_angs[1, 2]
+    po_angs_order[1, 0] = pol_angs[2, 0]
+    po_angs_order[2, 0] = pol_angs[2, 1]
+    po_angs_order[2, 1] = pol_angs[2, 2]
+
+    if symmetrize:
+        pol_au_order = 0.5 * (pol_au_order + pol_au_order.T)
+        po_angs_order = 0.5 * (po_angs_order + po_angs_order.T)
+
+    return pol_au_order, po_angs_order
+
+
+# function to calculate differential cross section
+def diff_cross_section(I_k, nu_k, laser_wl=0, T=300):
+    # defining universal constants
+    c = 299792458
+    mu_0 = 1.25663706212 * 1e-6
+    eps_0 = np.reciprocal(np.square(c) * mu_0)
+    k_B = 1.380649 * 1e-23
+    h = 6.62607015 * 1e-34
+    nm2cm = 1e-7
```

Comment:
To make this future-proof, it would be helpful to add the units as comments after each value.

Reply:
Done